testsuite/dynaconf_loader.py

```diff
@@ -109,24 +109,33 @@ def _apicast_image(ocp):
         return lookup.model.spec.template.spec.containers[0].image
 
 
-def _rhsso_password(server_url, token):
+def _rhsso_password(tools_config, rhsso_config):
     """Search for SSO admin password"""
+    server_url = tools_config.get("server")
+    project = tools_config.get("namespace")
+    token = tools_config.get("token")
+    kind = rhsso_config.get("kind", "rhbk")
     try:
         # is this RHOAM?
         tools = OpenShiftClient(project_name="redhat-rhoam-user-sso", server_url=server_url, token=token)
         return tools.secrets["credential-rhssouser"]["ADMIN_PASSWORD"].decode("utf-8")
     except (OpenShiftPythonException, KeyError):
         try:
             # was it deployed as part of tools?
-            tools = OpenShiftClient(project_name="tools", server_url=server_url, token=token)
-            # first RHSSO 7.5 way
-            return tools.secrets["credential-sso"]["ADMIN_PASSWORD"].decode("utf-8")
+            tools = OpenShiftClient(project_name=project, server_url=server_url, token=token)
+            if kind == "rhbk":
+                return tools.secrets["rhbk-initial-admin"]["password"].decode("utf-8")
+            if kind == "rhsso":
+                # first RHSSO 7.5 way
+                return tools.secrets["credential-sso"]["ADMIN_PASSWORD"].decode("utf-8")
         except (OpenShiftPythonException, KeyError):
             try:
                 # try 7.4 known deployment if previous fails
                 return tools.environ("dc/sso")["SSO_ADMIN_PASSWORD"]
             except (OpenShiftPythonException, KeyError):
-                return None
+                pass
+
+    return None
```

Comment:
This depth seems kinda redundant and deep, every positive branch ends with the return so nesting the code into the except branches is not good idea to be honest.